src/gromacs/nblib/gmxsetup.cpp

```diff
@@ -198,6 +198,31 @@ void NbvSetupUtil::unpackTopologyToGmx()
     }
 }
 
+void NbvSetupUtil::setAtomProperties(std::unique_ptr<nonbonded_verlet_t>& nbv)
+{
+    t_mdatoms mdatoms;
+    // We only use (read) the atom type and charge from mdatoms
+    mdatoms.typeA = const_cast<int*>(system_->topology().getParticleTypeIdOfAllParticles().data());
+    mdatoms.chargeA = const_cast<real*>(system_->topology().getCharges().data());
+    nbv->setAtomProperties(mdatoms, particleInfoAllVdw_);
```

Comment:
The actual setting in setAtomProperties() should be a separate function that takes the mdatoms constructed in its own function. The mdatoms need to be passed to the GmxForceCalculator anyway so this cannot be a temporary.

Reply:
Made `t_mdatoms` a member of `GmxForceCalculator` and made the created `mdatoms` persistent.